config/app.go

```diff
@@ -16,6 +17,56 @@ type ValidationError struct {
 	EnvVar []string
 }
 
+// Github Pull Request type.
+type PullRequest struct {
+	Host   string
+	Owner  string
+	Repo   string
+	Number int
+}
+
+// LoadFromURL parses Github Pull Request URL
+// into PullRequest object.
+func (p *PullRequest) LoadFromURL() error {
+	var (
+		err error
+	)
+
+	URL := os.Getenv(EnvCiCircleCiPullRequestID)
+
+	if URL == "" {
+		return err
+	}
+
+	u, err := url.Parse(URL)
+
+	if err != nil {
+		return err
+	}
+
+	p.Host = u.Host
+
+	path := strings.Split(u.Path, "/")
+	plen := len(path)
+
+	switch {
+	case plen == 2:
+		p.Number, err = strconv.Atoi(path[1])
+	case plen == 1:
+		p.Number, err = strconv.Atoi(path[0])
+	case plen == 4:
+		p.Owner = path[1]
+		p.Repo = path[2]
+		p.Number, err = strconv.Atoi(path[3])
+	case plen > 4:
```

Comment:
can you cover these scenarios in tests?

Reply:
Yes, working on it

Reply:
@mmogylenko im looking forward for your tests, I can not merge this until this is covered. It will also help me to understand the different scenarios